src/openrct2/ReplayManager.cpp

```diff
@@ -0,0 +1,775 @@
+                parkParams << gCheatsAllowArbitraryRideTypeChanges;
+                parkParams << gCheatsDisableRideValueAging;
+                parkParams << gConfigGeneral.show_real_names_of_guests;
+                parkParams << gCheatsIgnoreResearchStatus;
+
+                game_load_init();
+                fix_invalid_vehicle_sprite_sizes();
+            }
+            catch (const std::exception&)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        bool ReadReplayFromFile(const std::string& file, MemoryStream& stream)
+        {
+            FILE* fp = fopen(file.c_str(), "rb");
+            if (!fp)
+                return false;
+
+            char buffer[128];
+            while (feof(fp) == false)
+            {
+                size_t read = fread(buffer, 1, 128, fp);
+                if (read == 0)
+                    break;
+                stream.Write(buffer, read);
+            }
+
+            fclose(fp);
+            return true;
+        }
+
+        bool ReadReplayData(const std::string& file, ReplayRecordData& data)
+        {
+            MemoryStream stream;
+            DataSerialiser serialiser(false, stream);
+
+            std::string fileName = file;
+            if (fileName.size() < 5 || fileName.substr(fileName.size() - 5) != ".sv6r")
+            {
+                fileName += ".sv6r";
+            }
+
+            std::string outPath = GetContext()->GetPlatformEnvironment()->GetDirectoryPath(DIRBASE::USER, DIRID::REPLAY);
+            std::string outFile = Path::Combine(outPath, fileName);
+
+            bool loaded = false;
+            if (ReadReplayFromFile(outFile, stream))
+            {
+                loaded = true;
+            }
+            else if (ReadReplayFromFile(file, stream))
+            {
+                loaded = true;
+            }
+            if (!loaded)
+                return false;
+
+            stream.SetPosition(0);
+
+            if (!Serialise(serialiser, data))
+            {
+                return false;
+            }
+
+            // Reset position of all streams.
+            data.parkData.SetPosition(0);
+            data.parkParams.SetPosition(0);
+            data.spriteSpatialData.SetPosition(0);
+
+            return true;
+        }
+
+        bool SerialiseCommand(DataSerialiser& serialiser, ReplayCommand& command)
+        {
+            serialiser << command.tick;
+            serialiser << command.commandIndex;
+
+            bool isGameAction = false;
+            if (serialiser.IsSaving())
+            {
+                isGameAction = command.action != nullptr;
+            }
+            serialiser << isGameAction;
+
+            if (isGameAction)
+            {
+                uint32_t actionType = 0;
+                if (serialiser.IsSaving())
+                {
+                    actionType = command.action->GetType();
+                }
+                serialiser << actionType;
```

Comment:
When `serialiser.IsSaving()` (the condition above) returns false, will this statement alter the value of `actionType`?

Reply:
Yes